core/node/node_framework/src/implementations/layers/housekeeper.rs

```diff
@@ -0,0 +1,316 @@
+        context.add_task(Box::new(WaitingToQueuedFriWitnessJobMoverTask {
+            waiting_to_queued_fri_witness_job_mover,
+        }));
+
+        let scheduler_circuit_queuer = SchedulerCircuitQueuer::new(
+            self.house_keeper_config.fri_witness_job_moving_interval_ms,
+            prover_pool.clone(),
+        );
+        context.add_task(Box::new(SchedulerCircuitQueuerTask {
+            scheduler_circuit_queuer,
+        }));
+
+        let fri_witness_generator_stats_reporter = FriWitnessGeneratorStatsReporter::new(
+            prover_pool.clone(),
+            self.house_keeper_config
+                .witness_generator_stats_reporting_interval_ms,
+        );
+        context.add_task(Box::new(FriWitnessGeneratorStatsReporterTask {
+            fri_witness_generator_stats_reporter,
+        }));
+
+        let fri_prover_stats_reporter = FriProverStatsReporter::new(
+            self.house_keeper_config
+                .fri_prover_stats_reporting_interval_ms,
+            prover_pool.clone(),
+            replica_pool.clone(),
+            self.fri_prover_group_config,
+        );
+        context.add_task(Box::new(FriProverStatsReporterTask {
+            fri_prover_stats_reporter,
+        }));
+
+        let fri_proof_compressor_stats_reporter = FriProofCompressorStatsReporter::new(
+            self.house_keeper_config
+                .fri_proof_compressor_stats_reporting_interval_ms,
+            prover_pool.clone(),
+        );
+        context.add_task(Box::new(FriProofCompressorStatsReporterTask {
+            fri_proof_compressor_stats_reporter,
+        }));
+
+        let fri_proof_compressor_retry_manager = FriProofCompressorJobRetryManager::new(
+            self.fri_proof_compressor_config.max_attempts,
+            self.fri_proof_compressor_config.generation_timeout(),
+            self.house_keeper_config
+                .fri_proof_compressor_job_retrying_interval_ms,
+            prover_pool.clone(),
+        );
+        context.add_task(Box::new(FriProofCompressorJobRetryManagerTask {
+            fri_proof_compressor_retry_manager,
+        }));
+
+        Ok(())
+    }
+}
+
+struct PoolForMetricsTask {
+    pool_for_metrics: ConnectionPool,
+}
+
+#[async_trait::async_trait]
+impl Task for PoolForMetricsTask {
+    fn name(&self) -> &'static str {
+        "pool_for_metrics"
+    }
+
+    async fn run(self: Box<Self>, _stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        self.pool_for_metrics
+            .run_postgres_metrics_scraping(SCRAPE_INTERVAL)
+            .await;
+
+        Ok(())
+    }
+}
+struct L1BatchMetricsReporterTask {
+    l1_batch_metrics_reporter: L1BatchMetricsReporter,
+}
+
+#[async_trait::async_trait]
+impl Task for L1BatchMetricsReporterTask {
+    fn name(&self) -> &'static str {
+        "l1_batch_metrics_reporter"
+    }
+
+    async fn run(self: Box<Self>, _stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        self.l1_batch_metrics_reporter.run().await
```

Comment:
Looks like most of house keeper tasks are not cancellation-aware. It's OK to keep it as is for now, but could you please create a linear issue to add support of stop receiver to the house keeper tasks and then add TODO comments mentioning this task here?